block/block_test.go

```diff
@@ -3,16 +3,17 @@
 // Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
 // file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
 
-package block_test
+package block
```

Comment:
Suggestion: I think it might make sense to keep `block_test` so we also check the method visibility as of now.

Reply:
the go convention is to have the implementation and unit tests in the same package. If we want to have a set of tests interacting with the implementation via the public methods/structs exposed, they should be placed in a separated package named `block_test`

Reply:
Yep, just thought that it might make sense given the current status. Thanks.